common/flatpak-dir.c

```diff
@@ -11906,8 +11907,30 @@ add_subdirs (GPtrArray *res,
           info == NULL)
         break;
 
+      /* Here we support either a plain repo or, if @recurse is TRUE, the root
+       * directory of a USB created with "flatpak create-usb"
+       */
       if (g_file_info_get_file_type (info) == G_FILE_TYPE_DIRECTORY)
-        g_ptr_array_add (res, g_object_ref (path));
+        {
+          g_autoptr(OstreeRepo) repo = ostree_repo_new (path);
+
+          if (ostree_repo_open (repo, NULL, NULL))
+            g_ptr_array_add (res, g_object_ref (path));
+          else if (recurse)
+            {
+              g_autoptr(GFile) ostree_repo_subpath = g_file_resolve_relative_path (path, "ostree/repo");
+              g_autoptr(GFile) dot_ostree_repo_subpath = g_file_resolve_relative_path (path, ".ostree/repo");
+              g_autoptr(GFile) dot_ostree_repo_d_subpath = g_file_resolve_relative_path (path, ".ostree/repos.d");
+
+              if (ostree_repo_open (ostree_repo_subpath, NULL, NULL))
```

Comment:
Isn't this calling ostree_repo_open() on a GFile instead of an OstreeRepo. You need a ostree_repo_new() somewhere.

Reply:
Yes, my mistake